src/System.Threading.Tasks/tests/System.Runtime.CompilerServices/AsyncTaskMethodBuilderTests.cs

```diff
@@ -13,212 +13,206 @@ namespace System.Threading.Tasks.Tests
-                    SynchronizationContext.SetSynchronizationContext(previousContext);
-                }
-            }
+            // TrackedCount should increase as Create() is called, and decrease as SetResult() is called.
 
-            // AsyncTaskMethodBuilder<T>
-            {
-                // Creating a task builder, building it, completing it successfully
-                {
-                    var atmb = AsyncTaskMethodBuilder<int>.Create();
-                    var t = atmb.Task;
-                    Assert.True(t.Status == TaskStatus.WaitingForActivation, "    > FAILURE. Builder should still be active (ATMBT, build then set)");
-                    atmb.SetResult(43);
-                    Assert.True(t.Status == TaskStatus.RanToCompletion, "    > FAILURE. Builder should have successfully completed (ATMBT, build then set)");
-                    Assert.True(t.Result == 43, "    > FAILURE. Builder should completed with the set result (ATMBT, build then set)");
-                }
-
-                // Verify that AsyncTaskMethodBuilder<T> is not touching sync context
-                {
-                    SynchronizationContext previousContext = SynchronizationContext.Current;
-                    var trackedContext = new TrackOperationsSynchronizationContext();
-                    SynchronizationContext.SetSynchronizationContext(trackedContext);
+            // Completing in opposite order as created.
+            var avmb1 = AsyncVoidMethodBuilder.Create();
+            Assert.Equal(1, trackedContext.TrackedCount);
+            var avmb2 = AsyncVoidMethodBuilder.Create();
+            Assert.Equal(2, trackedContext.TrackedCount);
+            avmb2.SetResult();
+            Assert.Equal(1, trackedContext.TrackedCount);
+            avmb1.SetResult();
+            Assert.Equal(0, trackedContext.TrackedCount);
 
-                    var atmb = AsyncTaskMethodBuilder<string>.Create();
-                    Assert.True(trackedContext.TrackedCount == 0, "    > FAILURE. Builder should not interact with the sync context (before ATMBT set)");
-                    atmb.SetResult("async");
-                    Assert.True(trackedContext.TrackedCount == 0, "    > FAILURE. Builder should not interact with the sync context (after ATMBT set)");
+            // Completing in same order as created
+            avmb1 = AsyncVoidMethodBuilder.Create();
+            Assert.Equal(1, trackedContext.TrackedCount);
+            avmb2 = AsyncVoidMethodBuilder.Create();
+            Assert.Equal(2, trackedContext.TrackedCount);
+            avmb1.SetResult();
+            Assert.Equal(1, trackedContext.TrackedCount);
+            avmb2.SetResult();
+            Assert.Equal(0, trackedContext.TrackedCount);
 
-                    SynchronizationContext.SetSynchronizationContext(previousContext);
-                }
-            }
+            SynchronizationContext.SetSynchronizationContext(previousContext);
```

Comment:
Resetting the context should ideally be done in a finally block (here and elsewhere the pattern is employed).